[4.0] Media field fix

dgrammatiko opened the pull request:
Pull Request for Issue #30453 .

### Summary of Changes

- Bug fix
- Remove duplicate bindings

### Testing Instructions
- Download  and install the module from @Razzo1987
- Create a Swiper Slider module
- Go to "Options" tab
- Select a Background Image
- Try to select a different image
- Remove the image with the X button
- Try to select an image 
- add more fields and repeat the previous steps


### Actual result BEFORE applying this Pull Request

Modal doesn't open as expected

### Expected result AFTER applying this Pull Request

Modal  opens as expected

### Documentation Changes Required

Nope


PS a note here: 
- All Bootstrap init functions need a similar approach, so whenever a Bootstrap component is added programmatically it can be initialised.
- This is largely unneeded if all the components were custom elements. Custom elements have lifecycle events and therefore the initialisation can happen on the `connectedCallback`, no extra fuzz, they just work...

dgrammatiko committed:
fix

infograf768 commented on build/media_source/system/js/fields/joomla-field-media.w-c.es6.js:
```diff
@@ -192,11 +190,18 @@
+      const modalElement = this.querySelector(`#imageModal_${this.querySelector(this.input).id}`);
+
+      if (!button || !modalElement) {
+        throw new Error('Misconfiguaration...');
```
Misconfiguration... I guess

dgrammatiko committed:
typo, (well nothing new)

dgrammatiko committed:
Allow media fields in subforms

- Cleanup Media field
- Expose initilazation method of Bootstrpa Modlas to a Joomla.Bootstrap object
- Initilise Modlas on the lifecycle of the Custom Element

dgrammatiko committed:
Merge branch '4.0-dev_media_field' of github.com:dgrammatiko/joomla-cms into 4.0-dev_media_field

* '4.0-dev_media_field' of github.com:dgrammatiko/joomla-cms:
  typo, (well nothing new)

# Conflicts:
#	build/media_source/system/js/fields/joomla-field-media.w-c.es6.js

dgrammatiko committed:
Paleolithic js mode only in the legacy…

dgrammatiko committed:
CS

dgrammatiko committed:
Proper check

dgrammatiko committed:
Pffff

dgrammatiko committed:
Update joomla-field-media.w-c.es6.js

Quy commented:
I have tested this item :white_check_mark: successfully on 50a892acda4a8d677fe50d02b995c535ddc0e100<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30455">issues.joomla.org/tracker/joomla-cms/30455</a>.</sub>

Razzo1987 commented:
I'm testing with
```
Joomla! 4.0.0-beta4-dev+pr.30455
These packages were built based on commit 50a892acda4a8d677fe50d02b995c535ddc0e100 included in pull request 30455
These packages were built on Sun Aug 23 16:30:38 UTC 2020
```
from https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/30455/downloads/34914
and doesn't work, but I think it doesn't include your last edit :sob:

dgrammatiko commented:
@Razzo1987 try this one: https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/30455/downloads/34914/Joomla_4.0.0-beta4-dev+pr.30455-Development-Full_Package.zip

dgrammatiko commented:
@Razzo1987 Just realised that the last commit was not processed, ask someone with Drone access to restart the task

Razzo1987 commented:
I confirm https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/30455/downloads/34914/Joomla_4.0.0-beta4-dev+pr.30455-Development-Full_Package.zip have the same problem.
Attend for Drone - restart the task :+1:

Quy commented:
@Razzo1987 Drone restarted. Please try again.

Razzo1987 commented:
Doesn't work
```
Joomla! 4.0.0-beta4-dev+pr.30455
These packages were built based on commit 50a892acda4a8d677fe50d02b995c535ddc0e100 included in pull request 30455
These packages were built on Sun Aug 23 17:59:30 UTC 2020
```

dgrammatiko commented:
> Doesn't work

Well, the drone was restarted but there is no new zip file so basically you've just tested an earlier version of this PR

Quy commented:
It is working for me. It includes the latest commit. See:

> These packages were built based on commit 50a892acda4a8d677fe50d02b995c535ddc0e100

Please enable `Debug System` and check the browser's console.

Quy commented:
Also use the new download link: https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/30455/downloads/34916/

infograf768 commented:
I have tested this item :white_check_mark: successfully on 50a892acda4a8d677fe50d02b995c535ddc0e100<br /><br />Works fine.

Question: could this patch have an impact on other modals (xtd)?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30455">issues.joomla.org/tracker/joomla-cms/30455</a>.</sub>

dgrammatiko commented:
> Question: could this patch have an impact on other modals (xtd)?

I doubt it will affect anything, basically what was done here is to expose the init functionality so it can be used by devs whenever adding programmatically modals. This is needed because all the components using the Bootstrap js need initialisation but custom elements don't need that kind of pampering, which for me is an improved DX.

infograf768 commented:
Trust you. ;) RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30455">issues.joomla.org/tracker/joomla-cms/30455</a>.</sub>

Razzo1987 commented:
Sorry....
for me doesn't work both:
```
Joomla! 4.0.0-beta4-dev+pr.30455
These packages were built based on commit 50a892acda4a8d677fe50d02b995c535ddc0e100 included in pull request 30455
These packages were built on Sun Aug 23 17:59:30 UTC 2020
```
https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/30455/downloads/34916/

and
```
Joomla! 4.0
These packages were built based on commit 628eafdb6e2542918f316dd46d1c33accd5c1a20
These packages were last built: Tuesday, 25 August 2020 02:00:18 UTC
```
https://developer.joomla.org/nightly-builds.html

@infograf768 or @Quy which installation do you use?

infograf768 commented:
Tks!

Didldu-Florian commented:
In current BETA Problem still exists: See https://github.com/joomla/joomla-cms/issues/26711
and #28953

Didldu-Florian commented:
@dgrammatiko can you check again?

dgrammatiko commented:
@Didldu-Florian change https://github.com/joomla/joomla-cms/blob/705ce67c714bbcd364a3fcb7e13863fb4e5e4147/build/media_source/system/js/fields/joomla-field-media.w-c.es6.js#L103 to `&& window.bootstrap.Modal.getInstance(this.modalElement) === null) {`

Also if you do a PR please also apply the same fix here: https://github.com/joomla/joomla-cms/blob/705ce67c714bbcd364a3fcb7e13863fb4e5e4147/build/media_source/system/js/fields/joomla-field-user.w-c.es6.js#L60

Didldu-Florian commented:
@dgrammatiko oh, nice, thanks :) That fix works.
But why does no PR exists for that?
